### DISCUSSION TITLE
Value of using react-script-editor for site modernization

### DISCUSSION THREAD

**Discussion by kc-va541-web** [pnp/sp-dev-fx-webparts#1817] [General]
Whet's the value of deploying/allowing react-script-editor on modern sites if it basically recreates the deficiencies inherit with Classic Script Editor webpart.  Deploying this doesn't give IT control of what code ets deployed to a site, only who and where it gets deployed.  I dont see the value of using this or similar webpart that still allow the execution of arbitrary scripts. Doesn't this webpart still allow JavaScript injection and still have the same weakness or vulnerabilities as the old Script Editor webpart.  The only difference I see is that SPFx was used to create a workaround.  How is using this webpart making the organization better or safer?

**Comment by phillipharding** [comment]
This repository is an open-source community and the webpart, along with its _on-premises_ companion, was provided as a sample.

An organisation should perform it's own evaluation of the risks involved in deploying solutions into their environment, especially ones that are not **ScriptSafe** and come with the risks you highlighted.

The README does include a warning note...
>Notice. All client-side web parts are deployed or enabled to be available in site level by tenant administrator using tenant app catalog. If there are concerns on enabling script options in a tenant, this web part or a approach should not be approved by tenant administrators.

**Comment by PathToSharePoint** [comment]
In my opinion, the main value is to offer organizations an easy migration path to modern. You can lift and shift your current environment, including CEWP and SEWP, and later restructure your pages.

To Phil's point, this is just a sample and you can have your own flavor. I have built a CEWP replacement that I call the "Dangerous Content Web Part", to make people aware of the risks (the name was inspired by React). Then you can add your own sanitizers, such as HtmlSanitizer, to make it safer (although AFAIK you'll never get 100% protection, it's like a vaccine...).
